### PR TITLE
CON-258 Create controllers and model for bouquets

### PR DESCRIPTION
**Pull request by koitoror**
## Description ##
This pull request creates a model and controllers for bouquets.

## How can This be tested ##
- Clone the repo: `git clone https://github.com/andela/mrm_api.git`
- Setup project according to `Readme.md`
- checkout to branch `story/CON-258-create-bouquet-controller-and-model`
- to fetch all the bouquets on the database, hit the endpoint `v2/bouquets` with a get request
- to run migrations
  `alembic stamp base`
  `alembic upgrade head`
- to run tests and test coverage `pytest --cov api/v2`

## Type of change ##
Please select the relevant option
- [x] Bugfix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## JIRA ##
[CON-258](https://andela-apprenticeship.atlassian.net/browse/CON-258)

## Relevant screenshots: ##
Screenshot 2019-09-03 at 10 10 07

### CODE REVIEW

**Comment by joshuaocero** [comment]
@koitoror, I've since merged @mnswaleh's PR. Please fix conflicts and possible alembic issues.

**Comment by koitoror** [comment]
> @koitoror, I've since merged @mnswaleh's PR. Please fix conflicts and possible alembic issues.

Noted sir.

File: alembic/env.py [hunk 1]
```diff
@@ -51,15 +52,13 @@
 
 def run_migrations_offline():
     """Run migrations in 'offline' mode.
-
```
Comment (mnswaleh) [inline alembic/env.py]: This change is not required

File: alembic/env.py [hunk 2]
```diff
@@ -51,15 +52,13 @@
...
     This configures the context with just a URL
     and not an Engine, though an Engine is acceptable
     here as well.  By skipping the Engine creation
     we don't even need a DBAPI to be available.
 
     Calls to context.execute() here emit the given string to the
     script output.
-
```
Comment (mnswaleh) [inline alembic/env.py]: This change is also uncalled for

File: tests/test_bouquets/test_bouquets.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from tests.base import BaseTestCase
+
+
+class TestChannels(BaseTestCase):
+    def test_get_all_channels(self):
```
Comment (mnswaleh) [inline tests/test_bouquets/test_bouquets.py]: You have to rename this class and test method

**Review by mnswaleh — CHANGES_REQUESTED** [review]
Nicely done. I specifically like the implementation of the model. Kindly attend to the comments below

File: tests/test_bouquets/test_bouquets.py (hunk 3 above)
Comment (koitoror) [reply tests/test_bouquets/test_bouquets.py]: Resolved

File: alembic/env.py (hunk 2 above)
Comment (koitoror) [reply alembic/env.py]: Resolved

**Review by mnswaleh — APPROVED** [review]

**Comment by joshuaocero** [comment]
@koitoror I do not see the `refresh_url` field. Where did that go?

File: api/v2/models/bouquets/bouquets_model.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+class Bouquets(Base, Utility):
+    __tablename__ = 'bouquets'
+    id = Column(Integer, Sequence('bouquet_id_seq', start=1, increment=1), primary_key=True) # noqa
+    api_key1 = Column(String, nullable=False)
+    api_key2 = Column(String, nullable=False)
+    auth_credentials = Column(String, nullable=False)
+    bouquet_name = Column(String)
+    should_refresh = Column(Boolean, default=False)
+    state = Column(Enum(StateType), nullable=False, default="active")
+
```
Comment (joshuaocero) [inline api/v2/models/bouquets/bouquets_model.py:19]: @koitoror are we missing the `refresh_url` field somewhere?

File: api/v2/models/bouquets/bouquets_model.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        self.api_key1 = kwargs['api_key1']
+        self.api_key2 = kwargs['api_key2']
+        self.bouquet_name = kwargs['bouquet_name']
+        self.auth_credentials = kwargs['auth_credentials']
+        self.should_refresh = kwargs['should_refresh']
+
+    def refresh_channels(self):
+        """Method for registering channels"""
+        
+        # TODO: add more functionality
```
Comment (joshuaocero) [inline api/v2/models/bouquets/bouquets_model.py]: TODO comments usually describe better has to be done in this section. This makes the TODO listing easy to read. Please flesh out your TODO comments a bit more.

**Review by joshuaocero — CHANGES_REQUESTED** [review]
Great stuff, please take note of the changes requested.

File: api/v2/models/bouquets/bouquets_model.py (hunk 4 above)
Comment (koitoror) [reply api/v2/models/bouquets/bouquets_model.py:19]: Resolved. Thanks for noting.

File: api/v2/models/bouquets/bouquets_model.py (hunk 5 above)
Comment (joshuaocero) [reply api/v2/models/bouquets/bouquets_model.py]: @koitoror 👆

**Review by joshuaocero — APPROVED** [review]

**Pull request merged**